src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -0,0 +1,192 @@
+            });
+    }
+
+    public async Task<Either<ActionResult, Unit>> Delete(Guid releaseId, Guid keyStatisticId)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async release =>
+                await _persistenceHelper.CheckEntityExists<KeyStatistic>(keyStatisticId, query =>
+                    query
+                        .Include(keyStat => (keyStat as KeyStatisticDataBlock)!.DataBlock) // TODO Include can be removed in EES-3988
+                        .Where(keyStat => keyStat.ReleaseId == release.Id)))
+            // TODO: Can remove the below OnSuccess in EES-3988
+            // Key stats were previously associated with a KeyStatistic content section,
+            // which meant their data blocks had a ContentSectionId set. The below code
+            // ensures ContentSectionId is unset when a key stat is removed. If it isn't
+            // unset, that data block won't become available for selection again.
+            .OnSuccess(async keyStat =>
+            {
+                if (keyStat.GetType() == typeof(KeyStatisticDataBlock))
+                {
+                    var dataBlock = (keyStat as KeyStatisticDataBlock)!.DataBlock;
+                    dataBlock.ContentSection = null;
+                    dataBlock.ContentSectionId = null;
+                    _context.ContentBlocks.Update(dataBlock);
+                    await _context.SaveChangesAsync();
+                }
+                return keyStat;
+            })
+            .OnSuccessVoid(async keyStat =>
+            {
+                _context.KeyStatistics.Remove(keyStat);
+                await _context.SaveChangesAsync();
+            });
+    }
+
+    public async Task DeleteAnyAssociatedWithDataBlock(Guid releaseId, Guid dataBlockId)
+    {
+        var keyStats = await _context
+            .KeyStatisticsDataBlock
+            .Where(ks => ks.ReleaseId == releaseId && ks.DataBlockId == dataBlockId)
+            .ToListAsync();
+        _context.KeyStatisticsDataBlock.RemoveRange(keyStats);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task<Either<ActionResult, List<KeyStatisticViewModel>>> Reorder(
+        Guid releaseId,
+        Dictionary<Guid, int> newKeyStatisticOrder)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async release =>
+            {
+                var keyStatistics = _context.KeyStatistics
+                    .Where(ks => ks.ReleaseId == release.Id)
+                    .ToList();
+
+                var idOrderPairList = newKeyStatisticOrder.ToList();
+
+                if (keyStatistics.Count != idOrderPairList.Count)
+                {
+                    throw new ArgumentException("newKeyStatisticOrder.Count must equal release's key statistics count");
```

Comment:
Or alternatively `return VaildationUtils.ValidationActionResult(KeyStatsOrderCountShouldEqualReleaseKeyStatsCount);`

The advantage over throwing a generic ArgumentError is that this will result in a nicer API error response.